ui/src/app/state/user/user.reducer.ts

```diff
@@ -0,0 +1,16 @@
+import { Action, createReducer, on } from "@ngrx/store";
+import { User } from "../../services";
+import { setUser } from "./user.actions";
+
+export interface UserState {
+  user: User | undefined;
+}
+
+export const initialState: UserState = {
+  user: undefined
+};
+
+export const userReducer = createReducer<UserState, Action>(
+  initialState,
+  on(setUser, (_, { user }) => ({ user: user}))
```

Comment:
i know for this example this holds true, but to make clear it is common practice to return the full state in a sense we could also do
```suggestion
  on(setUser, (state, { user }) => ({  ...state, user }))
```